netket/sampler/autoreg.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2021 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import partial
+
+from netket.sampler import Sampler, SamplerState
+from netket.utils import struct
+from netket.utils.types import PRNGKeyT, PyTree
+
+import jax
+from jax import numpy as jnp
+
+
+def vmap_choice(key, a, p, replace=True):
+    """
+    p.shape: (batch, a.shape)
```

Comment:
can you add a bunch of comments? what is `a`

Reply:
and repliche

Reply:
I would like to rename it to `batch_choice`, as it does not really use `vmap` in the end. Also, I would not support output shapes of more than 1D, and there is no need to set `replace`.

Reply:
i completely approve.